### PR TITLE
Support account_host and check blob size attribute exists

### PR DESCRIPTION
**Pull request by KeerthiYandaOS**
Addresses the issue https://github.com/fsspec/adlfs/issues/340

### CODE REVIEW

**Comment by KeerthiYandaOS** [comment]
@hayesgb Can you please review the pull request? I'm unable to access the build logs, can you provide me access or steps to get the access? Thanks.

File: adlfs/spec.py [hunk 1]
```diff
@@ -584,10 +584,16 @@ def do_connect(self):
                 self.service_client = AIOBlobServiceClient.from_connection_string(
                     conn_str=self.connection_string
                 )
-            elif self.account_name:
-                self.account_url: str = (
-                    f"https://{self.account_name}.blob.core.windows.net"
-                )
+            elif self.account_name is not None:
+                if hasattr(self, 'account_host'):
```
Comment (TomAugspurger) [inline adlfs/spec.py:588]: Can you help me understand where this `account_host` attribute coming from? Is this from a subclass?

I'd rather have the `__init__` method accept something like `account_host` and make it a property on `AzureBlobFileSystem`. Then we can avoid all the hasattrs.

The interaction with the current `account_name` will be a bit complicated. Can you share an example of how account_host is being specified? Maybe we just don't allow users to pass both.

File: adlfs/spec.py [hunk 2]
```diff
@@ -892,7 +898,7 @@ async def _ls(
                                                 outblobs.append(blob_)
                                             elif (
                                                 blob_["name"].count("/") == depth
-                                                and blob_["size"] == 0
+                                                and (hasattr(blob_, "size") and blob_["size"] == 0)
```
Comment (TomAugspurger) [inline adlfs/spec.py:901]: Can you share a bit about why the `blob_` would have a size attribute? Does the lack of `size` attribute indicate a "directory" marker?

File: adlfs/spec.py (hunk 1 above)
Comment (KeerthiYandaOS) [reply adlfs/spec.py:588]: Adlfs is always appending the public blob endpoint. so having the ability to pass the account_host will help in accessing the non-public regions like gov regions. As account_host is not mandatory I was thinking of consumers to pass as required. We can also add it in __init__ and initialize it as None making as not a required property.

File: adlfs/spec.py (hunk 2 above)
Comment (KeerthiYandaOS) [reply adlfs/spec.py:901]: Yes Tom, have seen the blob not having size if the blob type is not a file.

**Pull request merged**